packages/sdg/src/lib/anchor-menu/anchor-menu.component.ts

```diff
@@ -0,0 +1,37 @@
+import { NgFor, NgIf } from '@angular/common';
+import { ChangeDetectionStrategy, Component, OnInit } from '@angular/core';
+import { ActivatedRoute, RouterLink } from '@angular/router';
+
+import { Anchor } from './anchor-menu.interface';
+
+@Component({
+  selector: 'sdg-anchor-menu',
+  standalone: true,
+  imports: [NgIf, NgFor, RouterLink],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  templateUrl: './anchor-menu.component.html',
+  styleUrls: ['./anchor-menu.component.scss']
+})
+export class AnchorMenuComponent implements OnInit {
+  constructor(private activatedRoute: ActivatedRoute) {}
+
+  anchors: Anchor[] = [];
+
+  ngOnInit() {
+    this.activatedRoute.fragment.subscribe((fragment) => {
+      if (fragment) {
+        this.jumpToSection(fragment);
+      }
+    });
+
+    const elements = document.getElementsByTagName('h2');
```

Comment:
Dans Angular c'est conseillé de ne pas accéder au document directement. Cela permet de générer les pages côtés serveurs (SSR).
Ici je te proposerais une méthode différente soit d'accéder au host de ta composante et de limiter la recherche à l'intérieur de celui-ci
```
constructor(
  private activatedRoute: ActivatedRoute,
  private elementRef: ElementRef
) {}

ngAfterContentInit(): void {
  ...
  const elements = (this.elementRef.nativeElement as HTMLElement).getElementsByTagName('h2');
  ...
}
```

Reply:
En intégrant ceci, `this.elementRef.nativeElement` retourne le html du composant et non de la page en entier. Est-ce qu'il faudrait plutôt que j'impante ceci dans la démo et qu'on passe `this.elementRef.nativeElement` en input au composant?

Reply:
Pour que ce soit fonctionnel la composante sdg-anchor-menu devrait englober le contenu et utiliser la projection via le ng-content

```
<sdg-anchor-menu>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod
    tempor incididunt ut labore et dolore magna aliqua. Lorem ipsum dolor sit
  </p>

  <h2 id="section1">Section 1</h2>

  ...
</sdg-anchor-menu>
```

Reply:
Ok je comprends mieux mainteant, mais il me semble que ça fait drôle quand même de mettre tout le contenu de la page dans le composant

Reply:
On pourrait p-t permettre à cette composante d'être au deux controllé et non-controllé. Si l'input anchors est détecté, on gère les valeurs de l'input sinon on recherche dans le contenu projeté les tags h2...?
Parce que la logique d'affaire de cette composante est toujours la même `seulement les tags h2 avec un id doit être utilisé comme "anchor"` alors je vois quand même une pertinence qu'elle soit non-controllé

Reply:
À mon avis, je le gérerais seulement par input finalement pour deux raisons:
1. Comme j'ai dit plus haut, je trouve ça bizarre de mettre tout le contenu de la page entre les balises `<sdg-anchor-menu>`. Plus clean d'avoir seulement la ligne:
![image](https://github.com/user-attachments/assets/439e48bb-316c-4635-a4b7-296fb6888545)
2. C'est mentionné que seulement des tags h2 peuvent être des anchors, mais pas nécessairement que TOUS les h2 DOIVENT être des anchors si le composant est présent. Donc en ayant en input on peut éventuellement créer un array `Anchor[]` manuellement en spécifiant les titres et les ids.